openflow_input/bsn_port_desc_module_eeprom

```diff
@@ -27,166 +27,10 @@
-    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_4G_2G_1G_AS0_AS1 = 0x01,              /* 4/2/1G Rate_Select & AS0/AS1 */
-    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_8G_4G_2G_RX_RATE_SELECT_ONLY = 0x02,  /* 8/4/2G Rx Rate_Select only */
-    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_8G_4G_2G_INDEPENDENT_RX_TX = 0x03,    /* 8/4/2G Independent Rx & Tx Rate_Select */
-    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_8G_4G_2G_TX_RATE_SELECT_ONLY = 0x04   /* 8/4/2G Tx Rate_Select only */
-};
-
-enum ofp_bsn_module_eeprom_cu_cmplnce(wire_type=uint8_t, bitmask=False) {
-    OFP_BSN_MODULE_EEPROM_CU_CMPLLNCE_UNSPECIFIED = 0x00,       /* unspecified */
-    OFP_BSN_MODULE_EEPROM_CU_CMPLLNCE_SFF_8431_APP_E = 0x01,    /* SFF-8431 appendix E */
-    OFP_BSN_MODULE_EEPROM_CU_CMPLLNCE_SFF_8431_LIMITING = 0x04  /* SFF-8431 limiting */
-};
-
-enum ofp_bsn_module_eeprom_alarm_set(wire_type=uint8_t, bitmask=True) {
-    OFP_BSN_MODULE_EEPROM_ALARM_NOT_SET = 0x00,         /* alarm not set for property */
-    OFP_BSN_MODULE_EEPROM_ALARM_HI_SET = 0x01,          /* high alarm set */
-    OFP_BSN_MODULE_EEPROM_ALARM_LOW_SET = 0x02,         /* low alarm set */
-    OFP_BSN_MODULE_EEPROM_ALARM_HI_WARN_SET = 0x04,     /* alarm set for high warning */
-    OFP_BSN_MODULE_EEPROM_ALARM_LOW_WARN_SET = 0x08,    /* alarm set for low warning */
-};
-
-struct of_port_desc_prop_bsn_alarm : of_port_desc_prop_bsn {
-    uint16_t    type == 0xffff;
-    uint16_t    length;
-    uint32_t    experimenter == 0x5c16c7;
-    uint32_t    exp_type == 8;
-    enum ofp_bsn_module_eeprom_alarm_set alarm_set; /* Track all alarms set for this property*/
-    uint32_t high;                                  /* Threshold for high */
-    uint32_t high_warn;                             /* Threshold for high warning */
-    uint32_t low;                                   /* Threshold for low */
-    uint32_t low_warn;                              /* Threshold for low warning */
-    enum ofp_bsn_module_eeprom_diag_type alarm_type;/* Diagnostic property for which the alarm is described */
-    enum ofp_bsn_unit_type unit;                    /* Unit for the high, low, high_warn and low_warn values */
-};
-
-struct ofp_bsn_module_eeprom_transceiver {
-    uint64_t    codes; /* The codes are in 8 bytes */
-};
-
-struct of_port_desc_prop_bsn_diag : of_port_desc_prop_bsn {
-    uint16_t    type == 0xffff;
-    uint16_t    length;
-    uint32_t    experimenter == 0x5c16c7;
-    uint32_t    exp_type == 7;
-    of_bsn_unit_t    laser_bias_curr;
-    of_bsn_unit_t    laser_output_power;
-    enum ofp_bsn_module_eeprom_recv_signal_type laser_receiver_power_type;
-    of_bsn_unit_t    laser_receiver_power;
-    of_bsn_unit_t    module_temp;
-    of_bsn_unit_t    module_voltage;
-};
-
 struct of_port_desc_prop_bsn_ethtool : of_port_desc_prop_bsn {
     uint16_t    type == 0xffff;
     uint16_t    length;
     uint32_t    experimenter == 0x5c16c7;
     uint32_t    exp_type == 6;
```

Comment:
do you want to keep the port description property name as "bsn_ethtool" ? or is there some more applicable name ?
we'll also need to update the "Port description property extensions" section of the openflow extensions wiki page.

Reply:
@kenchiang,
"do you want to keep the port description property name as "bsn_ethtool" ? or is there some more applicable name ?"

Do you mean some name which doesn't include bsn, and identifies what this is for?

Reply:
Summarizing the discussion I had with @kenchiang . Ken prefers that we have the name indicating what data it stores rather than something very generic like ethtool.

Reply:
The api's used to populate the data is sff_eeprom_parse,  sff_info_to_json, sff_dom_info_get,  sff_dom_info_to_json.

The data is a json, maybe we should have json in the of_port_desc_prop suffix.
The data is eeprom and dom info... throwing out some ideas:

of_port_desc_prop_bsn_eeprom_dom_json ?

Reply:
dom is  implicit, we could probably drop that from the name. 
If sff is widely used as compared to eeprom, we could probably stick to that.

Reply:
of_port_desc_prop_bsn_sff_info_json ?

Reply:
then of_port_desc_prop_bsn_sff_json?

Reply:
im ok with that.

Reply:
lets see what @kenchiang thinks.

Reply:
"of_port_desc_prop_bsn_sff_json" sounds good

Reply:
ok let me make the changes.